pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -180,6 +182,28 @@ func (c *TargetConfigController) getSubstitutionReplacer(operatorSpec *operatorv
 	), nil
 }
 
+func (c *TargetConfigController) checkDependencies() error {
+	// Ensure kube-controller-manager has rotated csr-controller-ca before rolling out.
+	// Waiting here is very important in single node as we want to avoid disruption until
+	// key resources have rotated.
+	csrControllerCAConfigMap, err := c.configMapLister.ConfigMaps(operatorclient.GlobalMachineSpecifiedConfigNamespace).Get("csr-controller-ca")
+	if err != nil {
+		return err
+	}
+
+	csrControllerCAConfigMapCopy := csrControllerCAConfigMap.DeepCopy()
+	var isCSRControllerCARotated bool
+	for _, managedField := range csrControllerCAConfigMapCopy.ManagedFields {
+		if managedField.Manager == "cluster-kube-controller-manager-operator" {
```

Comment:
I'm curious, is there a precedent for depending on ManagedField metadata to infer whether another agent in the cluster has performed some kind of operation on a resource? And is the presence of ckcmo as a field manager necessarily a guarantee of successful rotation? 

It's possible that this comment is just a reflection of my unease with the prevalence of the many 'API's' in OpenShift for which no explicit contract is published or agreed upon.

/cc @sttts @deads2k

Reply:
As the original resource was created by kube-apiserver-operator I believe KCM creating the resource would be indication of the signal required.

Reply:
I can review the key state to have a more concrete understanding of the signer so I will take that approach and see if the signal is more clear.